etcd-manager/pkg/controller/controller.go

```diff
@@ -950,13 +950,13 @@ func (m *EtcdController) removeNodeFromCluster(ctx context.Context, clusterSpec
 					peerState := m.peerState[privateapi.PeerId(id)]
 					if peerState == nil {
 						klog.Fatalf("peerState unexpectedly nil")
+					} else {
```

Comment:
This change seems less desirable than what was before.

Reply:
More or less agree, but `staticheck` was seeing a possible issue with the code. It's not so bad as it is now either.

Reply:
Any objection to merging this, now that this has been reverted?